### PR TITLE
Source Mailchimp: Add expected records

### LINKED ISSUE #32430: Source Mailchimp bypasses expected_records tests
### Topic

_No response_

### Relevant information

As part of Mailchimp's silver certification, all streams should return expected_records when possible. This requires a migration from our internal account to a sandbox account for CAT tests.

### PR DESCRIPTION
**Pull request by ChristoGrab**
## What

Currently, we are reading from our internal Mailchimp account for CAT tests, so expected_records can't be populated in order to avoid sharing internal data. We do have an old sandbox account that has now been reactivated and can be safely used for testing purposes.

## How

- Added expected_records for all streams other than Automations (need to upgrade to paid account to populate stream)
- Updated credentials in GSM
- Added missing enum value for `type` property in Campaigns stream
- In edge cases where a campaign has not yet received any clicks, the Reports stream returns an empty string for `clicks.last_click` property. This causes validation errors due to the declared `date-time` format. Added logic to exclude the property if it is an empty string.

### CODE REVIEW

**Review by cgardens — APPROVED** [review]
lgtm!

File: airbyte-integrations/connectors/source-mailchimp/source_mailchimp/streams.py [hunk 1]
```diff
@@ -277,6 +277,18 @@ class Reports(IncrementalMailChimpStream):
...
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+
+        response = super().parse_response(response, **kwargs)
+
+        # In some cases, the 'last_click' field is returned as an empty string,
+        # which causes validation errors on the `date-time` format.
+        # To avoid this, we remove the field if it is empty.
+        for record in response:
+            if record.get("clicks", {}).get("last_click") == "":
+                record["clicks"].pop("last_click", None)
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-mailchimp/source_mailchimp/streams.py:289]:
```suggestion
            clicks = record.get("clicks", {})
            if not clicks.get("last_click"):
                clicks.pop("last_click", None)
```

File: docs/integrations/sources/mailchimp.md [hunk 2]
```diff
@@ -76,7 +76,8 @@ Now that you have set up the Mailchimp source connector, check out the following
 
 | Version | Date       | Pull Request                                             | Subject                                                                    |
 |---------|------------|----------------------------------------------------------|----------------------------------------------------------------------------|
-| 0.8.0   | 2023-11-01 | [32032](https://github.com/airbytehq/airbyte/pull/32032) | Add ListMembers stream
+| 0.8.1   | 2023-11-06 | [32226](https://github.com/airbytehq/airbyte/pull/32226) | Add expected records                                                       |
```
Comment (lazebnyi) [inline docs/integrations/sources/mailchimp.md:79]:
```suggestion
| 0.8.1   | 2023-11-06 | [32226](https://github.com/airbytehq/airbyte/pull/32226) | Unmute expected records test after data anonymisation                                                       |
```

**Review by lazebnyi — APPROVED** [review]
LGTM
Minor comments left

**Review by evantahler — APPROVED** [review]
👍 Nice! I confirm that the expected records don't have PII (we made some fake test accounts)

**Pull request merged**